Userland/Applets/ClipboardHistory/ClipboardHistoryModel.cpp

```diff
@@ -119,13 +119,13 @@ ErrorOr<void> ClipboardHistoryModel::invalidate_model_and_file()
     return {};
 }
 
-void ClipboardHistoryModel::add_item(const GUI::Clipboard::DataAndType& item)
+void ClipboardHistoryModel::add_item(GUI::Clipboard::DataAndType const& item)
```

Comment:
That's a little bit weird to me that trimming the file is done by the GUI-related code.

Edit: I forgot that there is no clear decoupling here, but my point remains, as it should probably be done when reading the file and not when adding new items via the GUI.

Reply:
Yeah doing this in the GUI code felt weird to me too, but I didn't have a good idea for how to do it differently.

As for trimming when reading: I originally had it break out of the "reading in items" loop when the limit was reached. But that makes assumptions about the order of the items, which we don't really ensure. Then again this trimming also makes those assumptions. I think I need to introduce some sorting.

Reply:
> But that makes assumptions about the order of the items, which we don't really ensure.

Hmm, I know that we don't have timestamps or anything to really ensure it, but the underlying Vector is kept sorted. And then the JSON file is an exact copy of the vector. So, reading the X first lines in the files should do the trick.

Adding something more complex would just make us resilient to external people swapping lines in our file. I don't think it's worth the extra complexity.